daemon/container.go

```diff
@@ -326,6 +326,14 @@ func (container *Container) Start() (err error) {
 		return nil
 	}
 
+	if container.RemovalInProgress {
+		return fmt.Errorf("Container is already being removed. Can not start it.")
+	}
+
+	if container.Dead {
+		return fmt.Errorf("Container is already dead. Can not start it.")
```

Comment:
Can remove already, which is confusing because the user is calling start
Also, what does `dead` mean?